src/main/java/org/example/dao/UserDaoImpl.java

```diff
@@ -1,77 +1,73 @@
-        try (PreparedStatement statement = dataSource.getConnection().prepareStatement(sqlCommand)) {
-            statement.setObject(1, user.getId());
-            statement.setString(2, user.getName());
-            statement.setTimestamp(3, user.getCreationDate());
-            return statement.executeUpdate() == 1;
-        } catch (SQLException e) {
-            throw new RuntimeException(e);
-        }
+    @Transactional
+    public void save(User user) {
+        String sqlCommand = "INSERT INTO user_data (id, name, user_status, creation_date)" +
+                "VALUES (?, ?, ?::user_status, ?)";
+        jdbcTemplate.update(sqlCommand, user.getId(), user.getName(), user.getUserStatus().name(),
+                user.getCreationDate());
     }
 
     @Override
+    @Transactional
     public User get(UUID id) {
         String sqlCommand = "SELECT * FROM user_data WHERE id = ?";
-        try (PreparedStatement statement = dataSource.getConnection().prepareStatement(sqlCommand)) {
-            statement.setObject(1, id);
-            try (ResultSet resultSet = statement.executeQuery()) {
-                if (resultSet.next()) {
-                    UUID columnId = (UUID) resultSet.getObject("id");
-                    String columnName = resultSet.getString("name");
-                    Timestamp columnCreationDate = resultSet.getTimestamp("creation_date");
-                    User user = new User(columnName, columnCreationDate);
-                    user.setId(columnId);
-                    return user;
-                }
-            }
-        } catch (SQLException e) {
-            System.out.println(e);
-        }
-        return null;
+        return jdbcTemplate.queryForObject(sqlCommand, new UserRowMapper(), id);
     }
 
-    public boolean update(User user) {
-        String sqlCommand = "UPDATE user_data SET name = ?, creation_date = ? WHERE id = ?";
-        try (PreparedStatement statement = dataSource.getConnection().prepareStatement(sqlCommand)) {
-            statement.setString(1, user.getName());
-            statement.setTimestamp(2, user.getCreationDate());
-            statement.setObject(3, user.getId());
-            return statement.executeUpdate() == 1;
-        } catch (SQLException e) {
-            throw new RuntimeException(e);
+    @Override
+    @Transactional
+    public void updateUserStatusById(User user) {
+        String sqlCommand = "UPDATE user_data SET user_status = CAST(? AS user_status) WHERE id = ?";
+        jdbcTemplate.update(sqlCommand, user.getUserStatus().name(), user.getId());
+    }
+
+    @Override
+    @Transactional
+    public void updateUserAndCreateTicket(User user, Ticket ticket) {
+        if (!userUpdateCreateEnabled) {
+            throw new UnsupportedOperationException("Creating a ticket is disabled.");
         }
+        userDAO.updateUserStatusById(user);
+        ticketDao.save(ticket);
```

Comment:
this is bad practice:
1. do not call one dao method inside another dao method
2. do not call method from another DAO class

Instead of that use 2 separate SQLs in 1 method:
1. jdbcTemplate.update( update user )
2. jdbcTemplate.update(insert values in ticket table )

Reply:
Created new TicketService class, moved updateUserAndCreateTicket method into it